app/views/my_facilities/_bp_controlled_details_v2.html.erb

```diff
@@ -0,0 +1,106 @@
+<div class="row">
+  <div class="col-lg-12">
+    <div class="card card-responsive">
+      <div class="d-flex align-baseline jc-between">
+        <h3>
+          BP controlled
+        </h3>
+        <p class="c-grey-dark">
+          Last updated: <%= @last_updated_at %>
+        </p>
+      </div>
+      <p class="c-grey-dark">
+        <strong>What to look for:</strong> High percentages are good, and 70% or higher is ideal 
+      </p>
+      <div class="d-flex fw-wrap mb-16px">
+        <%= render "shared/size_and_zone_filters" %>
+      </div>
+      <% %w(large medium small community).each do |size| %>
+        <% next if @facilities_by_size[size].blank? %>
+        <p class="mt-20px mb-0px fw-bold">
+          <%= size.capitalize %> facilities
+        </p>
+        <table class="mt-3 mt-lg-4 table table-compact table-responsive-md table-hover analytics-table">
+          <colgroup>
+            <col>
+            <col>
+            <col>
+            <col class="table-divider">
+            <col>
+            <col class="table-divider">
+            <col>
+            <col class="table-divider">
+            <col>
+            <col class="table-divider">
+            <col>
+            <col class="table-divider">
+            <col>
+            <col class="table-divider">
+            <col>
+            <col class="table-divider">
+            <col>
+          </colgroup>
+          <thead>
+            <tr data-sort-method="thead" class="sorts">
+              <th class="row-label sort-label sort-label-small" style="width: 160px;">
+                Facilities
+              </th>
+              <th class="row-label sort-label sort-label-small" data-sort-method="number" style="width: 120px;">
+                Total assigned<br>patients
+              </th>
+              <th class="row-label sort-label sort-label-small" data-sort-method="number" style="width: 120px;">
+                Total registered<br>patients
+              </th>
+              <th class="row-label sort-label sort-label-small" colspan="2" style="width: 80px;">
+                6-month change
+              </th>
+              <% (@start_period..@period).each do |period| %>
+                <th class="row-label sort-label sort-label-small" data-sort-method="number" colspan="2">
+                  <%= period %>
+                </th>
+              <% end %>
+            </tr>
+          </thead>
+          <tbody>
+            <% @facilities_by_size[size].each do |facility| %>
+              <% six_month_rate_change = @data_for_facility[facility.name]["controlled_patients_rate"][@period] - @data_for_facility[facility.name]["controlled_patients_rate"][@start_period] || 0 %>
```

Comment:
Now that the `period` s here are way clearer, you could keep it inline. 

Long term we should really get this into a view component, will make this all much clearer.